example/config.cpp

```diff
@@ -515,14 +515,32 @@ std::shared_ptr<backend_config> config_data::get_backend_config(uint32_t backend
 	return nullptr;
 }
 
-io_pool_config config_data::get_io_pool_config(const std::string &pool_id) {
-	const io_pool_config default_config = {
+io_pool_config config_data::io_pool_defaults() {
+	io_pool_config sys_defaults = {
 		cfg_state.io_thread_num,
 		cfg_state.nonblocking_io_thread_num,
 		/*lifo*/ true,
 		/*queue_limit*/ 1000
 	};
 
+	const auto &root = parse_config()->root();
+	if (!root.has("options"))
+		return sys_defaults;
+
+	const auto &options = root["options"];
+	if (!options.has("io_pools"))
+		return sys_defaults;
+
+	const auto &io_pools = options["io_pools"];
+	if (!io_pools.has("defaults"))
+		return sys_defaults;
```

Comment:
I think stairs looks better.
```
if (root.has("options")) {
    const auto &options = root["options"];
    if (options.has("io_pools")) {
        const auto &io_pools = options["io_pools"];
        if (io_pools.has("defaults")) {
            return parse_io_pool_config(io_pools["defaults"], sys_defaults);
```
There is temptation of using exceptions here, note that it'll be consistent with file style.
```
try {
    auto &io_pools_defaults = parse_config()->root()["options"]["io_pools"]["defaults"];
    return parse_io_pool_config(io_pools_defaults, sys_defaults);
} catch (const config_error &exc) {
    return sys_defaults;
}
```

Reply:
I don't like stairs because they brings extra indentions that reduces available line length. Variant with exception looks more organic but let's do it in a separate PR.